app/models/story.rb

```diff
@@ -925,7 +925,10 @@ def send_referrer?
   end
 
   def set_domain_and_origin(domain_name)
-    domain_name&.sub!(/^www\d*\./, "")
+    if domain_name.present? && domain_name.split(".").size > 2 # only removes www\d* if the url is not like www10.org, issue #1339
+      domain_name&.sub!(/^www\d*\./, "")
```

Comment:
If the regexp were `domain_name&.sub!(/\Awww\d*\.(.+\.)/, "\1")` you wouldn't need the `if` wrapping this, it would express that we're only matching `www\d+` if there isn't another `.` segment. And it could use a test or two, that's a hairy bit of code!

Reply:
Thanks for the suggestion ! This is actually my very first contribution and i'm still new to Ruby and rails (I'm a French student learning software developpment). So any advice or suggestions are really helpful to me! 

The contribution is part of a team project for a course where we contribute to an open-source project. We are a group of six students from the French engineering school IMT Atlantique, and we chose to contribute to Lobsters. We'll do our best to help improve Lobsters for the next few months!

If you have any suggestions on how we could further contribute or specific areas that need attention, we would be glad to hear them. Currently, we're focusing on several issues tagged "good first issue," but we're open to other contributions as well.

Reply:
Well, I appreciate you helping out, and let me know if you need any paperwork signed off on.